### PR TITLE
add ui and  implement searching functionalities from the dummy datenbank

### LINKED ISSUE #7: Step 3
## Value Proposition
As a user I want to search for articles based on my own criteria
## Description
description_step1
### User should be able to :
- [ ] search for articles by thema (eg. politics, sport), sources (eg. politico, the new york times), the period when they published (from..to), countries
## Tasks
- [ ] add a form which includes text inputs for given criteria (thema, sources, period, countries) and a search button
- [ ] take the input values as parameters for the API key
- [ ] fetch data from the dummy database
- [ ] display results

### PR DESCRIPTION
**Pull request by anhthu253**
Please review

### CODE REVIEW

File: components/Button.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import styled from "styled-components";
 
-export default function Button({ children }) {
-  return <StyledButton>{children}</StyledButton>;
+export default function Button({ children, className }) {
+  return <StyledButton className={className}>{children}</StyledButton>;
```
Comment (sebsch1337) [inline components/Button.js:4]: Why do you combine classNames and styled components?

File: components/SearchContainer.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        name="publishedAt"
+        placeholder="eg. 2022-10-01"
+      />
+      <TextEntry
+        label="country"
+        id="country"
+        name="country"
+        placeholder="eg. england"
+      />
+      <StyledArticle>
```
Comment (sebsch1337) [inline components/SearchContainer.js:47]: Article tags are not used for formulars / inputs. You could use sections or divs.

File: components/TextEntry.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        id={id}
+        value={value}
+        name={name}
+        placeholder={placeholder}
+      ></StyledInput>
+    </StyledTextEntry>
+  );
+}
+
+const StyledTextEntry = styled.article`
```
Comment (sebsch1337) [inline components/TextEntry.js]: Same here :)

File: lib/db.js [hunk 4]
```diff
@@ -1,6 +1,10 @@
-const initialArticles = [
+const articles = [
   {
     id: "24343",
+    thema: "politics",
```
Comment (sebsch1337) [inline lib/db.js:4]: you could write "topic" instead of "thema" to stay english :)

File: pages/index.js [hunk 5]
```diff
@@ -1,14 +1,13 @@
...
+export default function Home() {
+  const [articles, setArticles] = useState([]);
 
-  return {
-    props: { articles: articles },
-  };
-}
-export default function Home({ articles }) {
+  useEffect(() => {
+    setArticles(JSON.parse(localStorage.getItem("articles")) ?? []);
```
Comment (sebsch1337) [inline pages/index.js]: You could use the localStorage hook that Thomas posted a few days ago. No need of useEffect on your index page then :)

**Review by sebsch1337 — CHANGES_REQUESTED** [review]
Wow, you're already far! But it's a lot for one pull. You could split it in API, search functionalities and show in UI parts.
Some little changes below. :)

File: components/Button.js (hunk 1 above)
Comment (anhthu253) [reply components/Button.js:4]: It is a trick i found in google. I was trying to style a react component with styled-component...(eg. const StyledButton = styled(Button)). But it resulted in no effect until className is included. Very strange

File: components/SearchContainer.js (hunk 2 above)
Comment (anhthu253) [reply components/SearchContainer.js:47]: Thank you!

File: components/TextEntry.js (hunk 3 above)
Comment (anhthu253) [reply components/TextEntry.js]: Thank you!

File: lib/db.js (hunk 4 above)
Comment (anhthu253) [reply lib/db.js:4]: yep hehe

File: pages/index.js (hunk 5 above)
Comment (anhthu253) [reply pages/index.js]: yeah i shall use it u re right. Thank you!

File: components/Button.js (hunk 1 above)
Comment (sebsch1337) [reply components/Button.js:4]: You could try const StyledButton = styled.button``;

This works well for me :)

Comment (anhthu253) [reply components/Button.js:4]: no but i want to style an already gestyled component. For example: I already styled the button in a Button component. Then in another page I use this component Button and want to add new styling to it. That s why.. styled(Button). styled.button`` is to style a html element

Comment (sebsch1337) [reply components/Button.js:4]: You could make a new button for bigger changes or props inside the styled components for smaller changes.

**Review by sebsch1337 — APPROVED** [review]

**Pull request merged**